### PR TITLE
[Merged by Bors] - feat: more lemmas about `List.dedup` and other lattice operations

### PR DESCRIPTION
**Pull request by eric-wieser**
Also the trivial `Multiset` wrappers.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/Dedup.lean
Comment (YaelDillies) [inline Mathlib/Data/List/Dedup.lean:138]: @Timeroot had this lemma somewhere. Did it ever land? What is the PR?

File: Mathlib/Data/List/Dedup.lean
Comment (YaelDillies) [inline Mathlib/Data/List/Dedup.lean:148]:
```suggestion
theorem Subset.dedup_append_left {xs ys : List α} (h : xs ⊆ ys) :
```
and can we have the right version stated using `List.Perm`?

File: Mathlib/Data/Multiset/FinsetOps.lean
Comment (YaelDillies) [inline Mathlib/Data/Multiset/FinsetOps.lean:294]: Same, can we have the right verSion?

File: Mathlib/Data/List/Dedup.lean
Comment (YaelDillies) [reply Mathlib/Data/List/Dedup.lean:148]: Hmm maybe, but the fact it's missing compared to the other direction is quite upsetting and might cause confusion in the future. At least add a comment explaining why the other lemma doesn't exist

Comment (eric-wieser) [reply Mathlib/Data/List/Dedup.lean]: I added the lemma, but in the multiset file where the proof was trivial.

File: Mathlib/Data/Multiset/FinsetOps.lean
Comment (eric-wieser) [reply Mathlib/Data/Multiset/FinsetOps.lean:294]: I don't think the right version is meaningfully true? What statement do you want?

Comment (YaelDillies) [reply Mathlib/Data/Multiset/FinsetOps.lean:294]: Ah indeed. But then this lemma is something like `Multiset.filter_true_of_mem h` except that `filter_true_of_mem` doesn't exist

File: Mathlib/Data/Multiset/Dedup.lean
Comment (YaelDillies) [inline Mathlib/Data/Multiset/Dedup.lean:154]:
```suggestion
/-- Note that the stronger `List.Subset.dedup_append_right` is proved earlier. -/
```

File: Mathlib/Data/Multiset/FinsetOps.lean
Comment (eric-wieser) [reply Mathlib/Data/Multiset/FinsetOps.lean:294]: It exists for `List` though; I golfed the proof there

File: Mathlib/Data/List/Dedup.lean
Comment (eric-wieser) [reply Mathlib/Data/List/Dedup.lean:138]: I assume we don't need to wait for an answer here? Of course, if @Timeroot comes along with a shorter proof after this is merged, we can always make a golf PR.

**Review by YaelDillies — APPROVED** [review]
Looks good now

maintainer merge

**Comment by eric-wieser** [comment]
maintainer merge

The conspiracy for bots to ignore Yael continues

**Comment by mattrobball** [comment]
!bench

**Comment by mattrobball** [comment]
I would guess the import change and the single simp are good but benched so I don't have to speculate. If it comes back reasonable, please merge. Thanks!

bors delegate+

**Pull request closed without merging**
